### ISSUE TITLE
Python 3.8b2 and TestDatetime64Timezone, too many warnings.

### ISSUE THREAD

**Issue by charris** [numpy/numpy#14077]
Running tests with python3.8b2+ on travis fails with two instead of one warning for TestDatetime64Timezone
```
E           AssertionError: 2 warnings found but 1 expected.
E           {message : DeprecationWarning('parsing timezone aware datetimes is deprecated; this will raise an error in the future'), category : 'DeprecationWarning', filename : '/home/travis/build/numpy/numpy/builds/venv/lib/python3.8/site-packages/numpy/core/tests/test_deprecations.py', lineno : 89, line : None}
E           {message : DeprecationWarning('an integer is required (got type float).  Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.'), category : 'DeprecationWarning', filename : '/home/travis/build/numpy/numpy/builds/venv/lib/python3.8/site-packages/numpy/core/tests/test_deprecations.py', lineno : 89, line : None}
```
I think this problem is on the Python end in the `timedelta` class. I note that the function on our end looks buggy also, assigning a long to an int. The function where this occurs is deprecated, but until it is removed we should fix it.

EDIT: we might just want to mark it `xfail(python > 3.7)` if the problem really is on the Python end, in which case it should be fixed at some point. But let's wait until the first or second 3.8 rc before taking that step.

### Reproducing code example:

```python
import numpy as np
<< your code here >>
```

**Comment by eric-wieser** [comment]
Is it `test_string` or `test_datetime` that emits these warnings?

**Comment by charris** [comment]
`numpy.core.tests.test_deprecations.TestDatetime64Timezone.test_datetime`. See https://travis-ci.org/numpy/numpy/jobs/564712088 .

**Comment by eric-wieser** [comment]
The failure is real, it's caused by these lines:
https://github.com/numpy/numpy/blob/058851c5cfc98f50f11237b1c13d77cfd1f40475/numpy/core/src/multiarray/datetime.c#L2249-L2280

https://github.com/numpy/numpy/blob/058851c5cfc98f50f11237b1c13d77cfd1f40475/numpy/core/src/multiarray/datetime.c#L2249-L2280

`total_seconds` returns a `float`, which python 3.8 requires us to explicitly round to an `int` rather than just calling `PyInt_AsLong`

**Comment by eric-wieser** [comment]
I didn't actually check that I fixed this - is Travis passing for 3.8 now?

**Comment by charris** [comment]
Yes.
